cubism-core-sys/build.rs

```diff
@@ -1,23 +1,25 @@
 use std::{env, path::PathBuf};
 
-fn main() {
-    println!("cargo:rerun-if-env-changed=CUBISM_CORE");
-    let target = env::var("TARGET").unwrap();
-    let (arch, vendor, sys, abi) = {
-        let mut target_s = target.split('-');
-        (
-            target_s.next().unwrap_or(""),
-            target_s.next().unwrap_or(""),
-            target_s.next().unwrap_or(""),
-            target_s.next().unwrap_or(""),
-        )
+fn lookup_cubism_core(arch: &str, vendor: &str, sys: &str, abi: &str, linking_strategy: &str) {
+    let mut lib_dir = if let Ok(lib_dir) = env::var("CUBISM_CORE").map(PathBuf::from) {
+        lib_dir
+    } else {
+        // TODO: this won't appear unless the compilation fails.
+        println!(
```

Comment:
I think we can just panic here as well, the compilation will fail at the linking stage anyways so panicking here probably causes cleaner output?

Reply:
Some people will prefer linking libraries manually (e.g. placing libraries to the default library directory, writing build.rs by their own) to specifying libraries by setting environment variables.

Reply:
ah yep makes sense